KeyboardHeightConstraint.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  KeyboardHeightConstraint.swift
+//  FuntastyKit-iOS
+//
+//  Created by Matěj Jirásek on 07/02/2018.
+//  Copyright © 2018 The Funtasty. All rights reserved.
+//
+
+import UIKit
+
+final class KeyboardHeightConstraint: NSLayoutConstraint {
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChange(_:)), name: .UIKeyboardWillChangeFrame, object: nil)
+    }
+
+    @objc
+    private func keyboardWillChange(_ notification: Notification) {
+        guard let userInfo = notification.userInfo else {
+            return
+        }
+
+        let duration = userInfo[UIKeyboardAnimationDurationUserInfoKey]
+            .flatMap { $0 as? NSNumber }
+            .flatMap { $0.doubleValue } ?? 0.0
+        let options = userInfo[UIKeyboardAnimationCurveUserInfoKey]
+            .flatMap { $0 as? NSNumber }
+            .flatMap { $0.uintValue << 16 }
+            .flatMap(UIViewAnimationOptions.init)
+        let height = userInfo[UIKeyboardFrameEndUserInfoKey]
+            .flatMap { $0 as? NSValue }
+            .flatMap { $0.cgRectValue.height } ?? 0.0
```

Comment:
I think that `flatMap`s on lines 26, 29, 30 and 33 could be plain `map`s since they are not called on any containers.

Reply:
Will look into it.

Reply:
Improved.